pkg/kube/kube_handler.go

```diff
@@ -843,6 +844,41 @@ func filterActiveNamespaces(namespaces []corev1.Namespace) []corev1.Namespace {
 	return readyNamespaces
 }
 
+// GetNamespacesFromList return the list of namespaces that the user has permission to access
+func (a *userHandler) GetNamespacesFromList(headerNamespaces []string) ([]corev1.Namespace, error) {
```

Comment:
In order to DRY this up could you just redefine `GetNamespaces` to accept an optional param like `namespaces []string`? If present, check the inputted list for those that are permitted; if not, fallback to current/existing behavior of "check all".

Reply:
Both functions are slightly different but certainly, we could merge them.

Reply:
Isn't this function the same as the existing one to filter namespaces for a user? Not sure why it's being re-defined nor why you need to do this filtering in your case (where the header defines the namespaces that the user can access - why can't you trust that?)

Reply:
I merged them